Glt 2025

Rufus125 opened the pull request:
Clears the database so the app does not crash on first start of the new GLT25 release.
Compare https://github.com/linuxtage/EventFahrplan/compare/glt-2022...Rufus125:glt-2025?expand=1 to see the changes

Commit:
Bumps version number, Updates the date of the GLT25 and URLs

johnjohndoe commented:
Please note that specifying the conference dates in build.gradle is no longer over needed: https://github.com/EventFahrplan/EventFahrplan/pull/641

Commit:
Removes dates as per https://github.com/linuxtage/EventFahrplan/pull/5#issuecomment-2221036753 they are not needed anymore

Rufus125 commented:
https://github.com/Rufus125/EventFahrplan/blob/glt-2022/database/src/main/java/info/metadude/android/eventfahrplan/database/sqliteopenhelper/SessionsDBOpenHelper.kt

vs

https://github.com/Rufus125/EventFahrplan/blob/glt-2025/database/src/main/java/info/metadude/android/eventfahrplan/database/sqliteopenhelper/SessionsDBOpenHelper.java#L167

johnjohndoe committed:
Ease understanding missing track color error.

johnjohndoe committed:
Use annotation v.1.8.2.

+ Changelog: https://developer.android.com/jetpack/androidx/releases/annotation#1.8.2

johnjohndoe committed:
Use lifecycle v.2.8.5.

+ Changelog: https://developer.android.com/jetpack/androidx/releases/lifecycle#2.8.5

johnjohndoe committed:
Use mockito v.5.13.0.

+ Changelog: https://github.com/mockito/mockito/compare/v5.12.0...v5.13.0

johnjohndoe committed:
Use junit-jupiter v.5.11.0.

+ Changelog: https://junit.org/junit5/docs/5.11.0/release-notes/

johnjohndoe committed:
Use kotlin v.2.0.20 & ksp v.2.0.20-1.0.25.

+ Changelogs:
  https://github.com/JetBrains/kotlin/releases/tag/v2.0.20
  https://github.com/google/ksp/releases/tag/2.0.20-1.0.25

johnjohndoe committed:
Use build-tools v.35.0.0.

johnjohndoe committed:
Use Jetpack Compose UI 2024.09.00.

+ Changelogs:
  https://developer.android.com/develop/ui/compose/bom/bom-mapping
  https://developer.android.com/jetpack/androidx/releases/compose-material3#1.3.0

johnjohndoe committed:
Use Android Gradle Plugin v.8.6.0.

+ Changelogs:
  https://developer.android.com/build/releases/past-releases/agp-8-6-0-release-notes
  https://developer.android.com/build/releases/gradle-plugin

johnjohndoe committed:
Use Gradle wrapper v.8.10.1

+ Changelogs:
  https://docs.gradle.org/8.10.1/release-notes.html
  https://docs.gradle.org/8.10/release-notes.html

johnjohndoe committed:
Speed up builds, reduce load for maven repositories.

+ Ref: https://github.com/actions/setup-java?tab=readme-ov-file#caching-packages-dependencies

johnjohndoe committed:
Merge pull request #672 from EventFahrplan/project-updates

Update project setup (kotlin 2.0.20, ksp 2.0.20-1.0.25, build-tool 35.0.0, Compose UI 2024.09.00, AGP 8.6.0, Gradle wrapper 8.10.1).

johnjohndoe committed:
Replace deprecated Modifier#animateItemPlacement with Modifier#animateItem.

johnjohndoe committed:
Remove unneeded opt-in.

johnjohndoe committed:
Merge pull request #673 from EventFahrplan/random-housekeeping

Random housekeeping

johnjohndoe committed:
Suppress Lint warning.

+ Warning: "Non-public primary constructor is exposed via the generated
 'copy()' method of the 'data' class."

johnjohndoe committed:
Use Jetpack Compose UI 2024.09.02.

+ Changelogs:
  https://developer.android.com/develop/ui/compose/bom/bom-mapping
  https://developer.android.com/jetpack/androidx/releases/compose-material3#1.3.0

johnjohndoe committed:
Use threetenbp v.1.7.0.

+ Changelog: https://www.threeten.org/threetenbp/changes-report.html#a1.7.0

johnjohndoe committed:
Use coroutines v.1.9.0.

+ Changelog: https://github.com/Kotlin/kotlinx.coroutines/releases/tag/1.9.0

johnjohndoe committed:
Use Android Gradle Plugin v.8.6.1.

+ Changelogs:
  https://developer.android.com/build/releases/past-releases/agp-8-6-0-release-notes
  https://developer.android.com/build/releases/gradle-plugin
  https://developer.android.com/studio/releases/fixed-bugs/studio/2024.2.2

johnjohndoe committed:
Merge pull request #674 from EventFahrplan/project-updates

Update project setup (Compose UI 2024.09.02, AGP 8.6.1, threetenbp 1.7.0, coroutines 1.9.0).

johnjohndoe committed:
Apply automatic formatting to schedule layout files.

+ Add comment to "times_layout".

johnjohndoe committed:
Indicate progress of horizontal scrolling above room names.

+ This is intended to help users to understand how many rooms are hidden
  off-screen and to encourage horizontal scrolling.
+ As the user scrolls from room to room, the progress line moves by a
  corresponding amount. This shows where the current room is in relation
  to the full width of all rooms.
+ Using progress indicator dots has been considered but discarded
  because with the amount of rooms there was no way to place that many
  dots while improving the user experience.

Resolves #532

johnjohndoe committed:
Clean up code.

johnjohndoe committed:
Fix broken screen navigation when tapping session alarm item.

+ When the device was rotated while the alarms screen was visible then
  tapping the session alarm resulted in no action.
+ The onSessionItemClickListener referenced in the ScreenNavigation lambda
  was null once the fragment was recreated.
+ Broken since: 0d3b128a665906df6d34d59b91f3ef6846e34f03

johnjohndoe committed:
Fix broken screen navigation when tapping session change item.

+ When the device was rotated while the alarms screen was visible then
  tapping the session change resulted in no action.
+ The onSessionListClickListener referenced in the ScreenNavigation lambda
  was null once the fragment was recreated.
+ Broken since: f3e530841118a64da1bc473d3a5712d811593848

johnjohndoe committed:
Merge pull request #676 from EventFahrplan/screen-navigation

Fix broken screen navigation.

johnjohndoe committed:
Hide progress line while progress bar is still animating.

+ The MainActivity hosts a progressBar view which is displayed when schedule
  data is fetched or parsed. While the progressBar is animating, the progress
  line in the FahrplanFragment is invisible. This helps to avoid visual clutter
  because the progress line is placed directly below the progress bar.
+ Wait a few milliseconds until the progress bar is really gone.

johnjohndoe committed:
Merge pull request #675 from EventFahrplan/horizontal-scrolling-progress-bar

Indicate progress of horizontal scrolling above room names.

johnjohndoe committed:
Apply automatic formatting.

johnjohndoe committed:
Use mockito v.5.14.2.

+ Changelog: https://github.com/mockito/mockito/compare/v5.13.0...v5.14.2

johnjohndoe committed:
Use junit-jupiter v.5.11.3.

+ Changelogs:
  https://junit.org/junit5/docs/5.11.3/release-notes/
  https://junit.org/junit5/docs/5.11.2/release-notes/
  https://junit.org/junit5/docs/5.11.1/release-notes/

johnjohndoe committed:
Use android-test-* v.1.6.0.

+ Changelog: https://github.com/mannodermaus/android-junit5/releases/tag/instrumentation-1.6.0

johnjohndoe committed:
Use lifecycle v.2.8.7.

+ Changelogs:
  https://developer.android.com/jetpack/androidx/releases/lifecycle#2.8.7
  https://developer.android.com/jetpack/androidx/releases/lifecycle#2.8.6

johnjohndoe committed:
Use kotlin v.2.0.21.

+ Changelog: https://github.com/JetBrains/kotlin/releases/tag/v2.0.21

johnjohndoe committed:
Use Jetpack Compose UI 2024.10.01.

+ Changelogs:
  https://developer.android.com/develop/ui/compose/bom/bom-mapping
  https://developer.android.com/jetpack/androidx/releases/compose-material3#1.3.1

johnjohndoe committed:
Use Android Gradle Plugin v.8.7.2.

+ Changelogs:
  https://developer.android.com/build/releases/past-releases/agp-8-7-0-release-notes
  https://developer.android.com/build/releases/gradle-plugin

johnjohndoe committed:
Use Gradle wrapper v.8.10.2.

+ Changelogs:
  https://docs.gradle.org/8.10.2/release-notes.html

johnjohndoe committed:
Apply automatic formatting.

johnjohndoe committed:
Allow customizing body text color of schedule changes dialog.

johnjohndoe committed:
Replace deprecated options menu API with MenuProvider.

Resolves #507

johnjohndoe committed:
Add search functionality.

+ Add "Search" menu item to toolbar / overflow menu.
+ The search screen opens as a separate screen on smartphones and as
  a sidepane view on tablets.
+ Provide a full text search across all Session properties, see SearchQueryFilter.
+ Search results appear immediately while typing the query.
+ Support to clear the search query.
+ Support to navigate to details screen when tapping a Session search result.
+ Use empty state illustration.

Relates to #230
Resolves #3

johnjohndoe committed:
Fix error message in AlarmsFragment.

johnjohndoe committed:
Protect users from having to navigate back through multiple identical fragments.

+ Affects tablet layouts only.
+ This limits the here refactored fragments from being added to the
  backstack more than once.
+ This should make navigation much more pleasent.

Resolves #9

johnjohndoe committed:
Add missing sidePane flag to make fragment code consistent.

+ No idea if this has any effect.

johnjohndoe committed:
Ease testing when ContentDescriptionFormatter is injected.

johnjohndoe committed:
Fix test function name.

johnjohndoe committed:
Update Portuguese (Brazil) translation.

+ Thank you @Smarzaro.

johnjohndoe committed:
Merge pull request #677 from EventFahrplan/portuguese-brazil-translation

Update Portuguese (Brazil) translation.

johnjohndoe committed:
Reduce repetition by extracting FormattingDelegate/DateFormatterDelegate.

johnjohndoe committed:
Reduce repetition by extracting PendingIntentDelegate/PendingIntentProvider.

johnjohndoe committed:
Temporary disable flaky FahrplanViewModelTest.

+ See https://github.com/EventFahrplan/EventFahrplan/issues/526
+ Related: ac5cba5e10f94553c314444247dd659a5f079339

johnjohndoe committed:
Merge pull request #678 from EventFahrplan/random-housekeeping

Random housekeeping

johnjohndoe committed:
Merge pull request #679 from EventFahrplan/menuprovider

Replace deprecated options menu API with MenuProvider.

johnjohndoe committed:
Use ksp v.2.0.21-1.0.26.

+ Changelogs:
  https://github.com/google/ksp/releases/tag/2.0.21-1.0.26

johnjohndoe committed:
Use annotation v.1.9.1.

+ Changelogs:
  https://developer.android.com/jetpack/androidx/releases/annotation#1.9.1
  https://developer.android.com/jetpack/androidx/releases/annotation#1.9.0

johnjohndoe committed:
Use turbine v.1.2.0.

+ Changelog: https://github.com/cashapp/turbine/releases/tag/1.2.0

johnjohndoe committed:
Document compileSdk requirement for core-ktx.

+ Changelog: https://developer.android.com/jetpack/androidx/releases/core#1.15.0

johnjohndoe committed:
Use constraintlayout v.2.2.0.

+ Changlogs:
  https://developer.android.com/jetpack/androidx/releases/constraintlayout#constraintlayout-2.2.0
  https://github.com/androidx/constraintlayout/wiki/What's-New-in-2.2

johnjohndoe committed:
Fix partically broken discovery of some artifacts by gradle-versions-plugin.

johnjohndoe committed:
Merge pull request #680 from EventFahrplan/project-updates

Update project setup (Compose UI 2024.10.01, AGP 8.7.2, kotlin 2.0.21, Gradle wrapper 8.10.2, constraintlayout 2.2.0).

johnjohndoe committed:
Add search history.

+ Up to 20 unique search queries are remembered.
+ The latest search query is presented at the top of the list.
+ Search queries can be deleted to protect users' privacy.

Relates to #3

johnjohndoe committed:
Merge pull request #681 from EventFahrplan/search-functionality

Add search functionality.

johnjohndoe committed:
Adjust line spacing of copyright logo text in About screen.

+ Affects longer text which breaks into multiple lines.

johnjohndoe committed:
Hide app name in portrait mode to make room for toolbar icons.

johnjohndoe committed:
Show search toolbar icon prominently in toolbar.

johnjohndoe committed:
Merge pull request #682 from EventFahrplan/random-housekeeping

Random housekeeping (Copyright logo text, search icon)

johnjohndoe committed:
Bump application version to v.1.67.0.

johnjohndoe committed:
Removed unused string.

+ The ContentDescriptionFormatter already references the correct strings.

johnjohndoe committed:
Translate "Search" strings to German.

johnjohndoe committed:
Fix crash happening at layout time.

+ "IllegalStateException: Fragment FahrplanFragment did not return a View
  from onCreateView() or this was called before onCreateView()."

johnjohndoe committed:
Translate "Search" strings to Portuguese (Brazil).

+ Thank you @Smarzaro.

johnjohndoe committed:
Improve accessibility of back button.

johnjohndoe committed:
Merge pull request #683 from EventFahrplan/random-refinements

Random refinements (translations, a11y).

johnjohndoe committed:
Fix content description for session without speaker names.

+ Previously, the correct text for zero speakers was never read out when
  using TalkBalk. The quantity string system of Android is not intended to
  be used without a digit in the actual text.
+ This affects all screens where sessions are shown: schedule screen,
  favorites, alarms, schedule changes, search results.
+ Ref: https://developer.android.com/guide/topics/resources/string-resource#Plurals
+ Thanks to @rotrot for immediate the Polish translation.

johnjohndoe committed:
Update Turkish translation.

+ Thank you Oguz Yuksel.

johnjohndoe committed:
Merge pull request #685 from EventFahrplan/translations-turkish

Update Turkish translation.

johnjohndoe committed:
Merge pull request #684 from EventFahrplan/plurals-quantity-zero

Fix content description for session without speaker names.

johnjohndoe committed:
Update Polish translation.

+ Thank you @rotrot!

johnjohndoe committed:
Merge pull request #686 from EventFahrplan/translations-polish

Update Polish translation.

johnjohndoe committed:
Use Fragment#content extension function to clean up code.

+ Add androidx.fragment:fragment-compose:1.8.5
  https://developer.android.com/jetpack/androidx/releases/fragment

johnjohndoe committed:
Merge pull request #687 from EventFahrplan/fragment-compose

Use Fragment#content extension function to clean up code.

johnjohndoe committed:
Use Gradle wrapper v.8.11.1.

+ Changelogs:
  https://docs.gradle.org/8.11.1/release-notes.html
  https://docs.gradle.org/8.11/release-notes.html

johnjohndoe committed:
Use engelsystem v.9.2.0.

+ Changelog: https://github.com/johnjohndoe/engelsystem/blob/master/CHANGELOG.md#v920

+ Updates in the dependency tree:

  info.metadude.kotlin.library.engelsystem:engelsystem-base:9.2.0
  +--- com.squareup.moshi:moshi:1.15.1
  +--- com.squareup.okhttp3:okhttp:4.12.0
  +--- com.squareup.okhttp3:logging-interceptor:4.12.0
  +--- com.squareup.retrofit2:retrofit:2.11.0
  +--- com.squareup.retrofit2:converter-moshi:2.11.0
  +--- org.threeten:threetenbp:1.7.0
  \--- org.jetbrains.kotlin:kotlin-stdlib:2.0.21

johnjohndoe committed:
Use ksp v.2.0.21-1.0.28.

+ Changelogs:
  https://github.com/google/ksp/releases/tag/2.0.21-1.0.28
  https://github.com/google/ksp/releases/tag/2.0.21-1.0.27

johnjohndoe committed:
Use unmockplugin v.0.9.0.

+ Changelog: https://github.com/bjoernQ/unmock-plugin/releases/tag/v0.9.0

johnjohndoe committed:
Use Jetpack Compose UI 2024.11.00.

+ Changelogs:
  https://developer.android.com/develop/ui/compose/bom/bom-mapping
  https://developer.android.com/jetpack/androidx/releases/compose-material3#1.3.1

johnjohndoe committed:
RIP JEV 2022.

+ See: https://events.ccc.de/2022/11/28/dezentral-2022/
+ Related: 3d7dfe54dc5f1ef07e3e2534e8243d7d155a7727

johnjohndoe committed:
Add 38C3 2024.

+ See: https://events.ccc.de/category/38c3/

johnjohndoe committed:
Allow to customize text sizes on session card.

johnjohndoe committed:
Fit title of 5 minutes talks into session card.

Relates to #570

johnjohndoe committed:
Fit more text on session card.

+ Reduce text size while still being readable.

Relates to #570

johnjohndoe committed:
Merge pull request #688 from EventFahrplan/session-card-layout

Update session card layout (text size)

johnjohndoe committed:
Merge pull request #689 from EventFahrplan/rip-jev-2022

RIP JEV 2022.

johnjohndoe committed:
Merge pull request #690 from EventFahrplan/project-updates

Update project setup (Jetpack Compose UI 2024.11.00, engelsystem 9.2.0, ksp 2.0.21-1.0.28, unmockplugin 0.9.0).

johnjohndoe committed:
Simplify background color assignment for session details screen.

johnjohndoe committed:
Ease customizing light/dark mode for alarms, schedule changes and search screens.

+ Set to light theme for all three screens.
+ Schedule changes was dark before.
+ The theme of the details and the favorites screen has to configured
  separately. See colors/session_details_background and
  colors/favorites_background.

johnjohndoe committed:
Merge pull request #691 from EventFahrplan/color-customization

Simplify color customization for alarms, schedule changes, search and details screens.

johnjohndoe committed:
Reorganize alarms screen related colors resources.

johnjohndoe committed:
Apply system light/dark theme via Material3 theme to composified screens.

+ Affects alarms, schedule changes, search, about and schedule statistics screens.
+ These screens now automatically switch to light/dark mode dependent on
  the mode set in the device.
+ Add TODOs for app domain specific colors which can be moved into an
  app theme later.
+ Use MultiDevicePreview to preview light/dark mode at the same time.
+ Drop colors which are now provided via the theme.

johnjohndoe committed:
Customize schedule change colors for light/dark mode.

+ Colors needed some adjustments on light background.
+ Add TODO for change state colors which can be moved into an app
  theme later.

johnjohndoe committed:
RIP 37C3 2023.

+ See: https://events.ccc.de/category/37c3/
+ Related: 51bc4c7e45e491790fb2dc81208a8367cab65992

johnjohndoe committed:
Merge pull request #692 from EventFahrplan/material-theme-color-scheme

Apply system light/dark theme via Material3 theme to composified screens.

johnjohndoe committed:
Update Turkish translation.

+ Thank you Oguz Yuksel.

johnjohndoe committed:
Update Polish translation.

+ Thank you @rotrot!

johnjohndoe committed:
Clear database from Camp 2023 & 37C3 on upgrade.

+ Version 1.61.0 (96) was the last Camp 2023 release on
  Google Play and F-Droid.
  Related: 6a41478cd2bd33640620d96d08e6adbea14b87cf

+ Version 1.67.0 (105) was the last 37C3 release on
  Google Play and F-Droid.
  Related: 59d668bc83165981a3e050981c0f09589cbedca0.

johnjohndoe committed:
Bump application version to v.1.68.0.

johnjohndoe committed:
Merge pull request #694 from EventFahrplan/translations

Update Turkish and Polish translation.

johnjohndoe committed:
Merge pull request #693 from EventFahrplan/rip-37c3

RIP 37C3 2023.

johnjohndoe committed:
Merge pull request #695 from EventFahrplan/ccc38c3

Add ccc38c3 product flavor / wipe database.

cketti committed:
Create new activity task when launching navigation

johnjohndoe committed:
Increase informative value shown in crash report.

+ This aims to add the exception type to the stacktrace of a crash report
  to provide more insight into the error.
+ Error reports from Google Play and Tracedroid lack the actual error
  message. So, the custom exception type will make error analysis
  easier.

johnjohndoe committed:
Merge pull request #704 from cketti/navigation_intent_flags

Create new activity task when launching navigation

johnjohndoe committed:
Update copyright notices.

johnjohndoe committed:
Fix crash at reboot (OnBootReceiver).

+ Error:

  RuntimeException: Unable to start receiver nerd.tuxmobil.fahrplan.congress
    .system.OnBootReceiver: kotlin.UninitializedPropertyAccessException:
    lateinit property alarmsDatabaseRepository has not been initialized
  Caused by: kotlin.UninitializedPropertyAccessException: lateinit property
    alarmsDatabaseRepository has not been initialized
    at nerd.tuxmobil.fahrplan.congress.repositories.AppRepository.readAlarms(SourceFile:710)

johnjohndoe committed:
Use junit-jupiter v.5.11.4.

+ Changelog: https://junit.org/junit5/docs/5.11.4/release-notes/

johnjohndoe committed:
Use Gradle wrapper v.8.12.

+ Changelog: https://docs.gradle.org/8.12/release-notes.html

johnjohndoe committed:
Use mockito v.5.15.2.

+ Changelog: https://github.com/mockito/mockito/compare/v5.14.2...v5.15.2

johnjohndoe committed:
Use moshi v.1.15.2.

+ Changelog: https://github.com/square/moshi/blob/master/CHANGELOG.md#version-1152

johnjohndoe committed:
Use coroutines v.1.10.1.

+ Changelogs:
  https://github.com/Kotlin/kotlinx.coroutines/releases/tag/1.10.1
  https://github.com/Kotlin/kotlinx.coroutines/releases/tag/1.10.0

johnjohndoe committed:
Use kotlin v.2.1.0 & ksp v.2.1.0-1.0.29.

+ Changelogs:
  https://github.com/JetBrains/kotlin/releases/tag/v2.1.0
  https://github.com/google/ksp/releases/tag/2.1.0-1.0.29
  https://github.com/google/ksp/releases/tag/2.1.0-1.0.28

johnjohndoe committed:
Use Android Gradle Plugin v.8.8.0.

+ Changelogs:
  https://developer.android.com/build/releases/past-releases/agp-8-8-0-release-notes
  https://developer.android.com/build/releases/gradle-plugin

johnjohndoe committed:
Use Jetpack Compose UI 2024.12.01.

+ Changelogs:
  https://developer.android.com/develop/ui/compose/bom/bom-mapping
  https://developer.android.com/jetpack/androidx/releases/compose-material3#1.3.1

johnjohndoe committed:
Workaround to finish hanging instrumentation tests.

+ Source: https://github.com/ReactiveCircus/android-emulator-runner/issues/385

johnjohndoe committed:
Enable usage metrics and send them to Google Play.

johnjohndoe committed:
Merge pull request #711 from EventFahrplan/github-actions-hanging-instrumentation-tests

Workaround to finish hanging instrumentation tests (GitHub Actions).

johnjohndoe committed:
Merge pull request #710 from EventFahrplan/OnBootReceiver-UninitializedPropertyAccessException

Fix crash at reboot (OnBootReceiver).

johnjohndoe committed:
Document sonarqube-gradle-plugin build error.

johnjohndoe committed:
Merge pull request #712 from EventFahrplan/project-updates

Update project setup (AGP 8.8.0, Jetpack Compose UI 2024.12.01, kotlin 2.1.0, coroutines 1.10.1, Gradle 8.12).

johnjohndoe committed:
Prevent headline flickering when session content loads slowly.

johnjohndoe committed:
Protect users from reading identical texts twice.

+ Trims the "description" text if it starts like the "abstract" text.
+ Related: 39f9eba1332d428e3ecb2211a9a7feb28786f2d1.

johnjohndoe committed:
Remove leading and trailing whitespace.

johnjohndoe committed:
Update Lithuanian translation.

+ Thank you zadintuvas!

johnjohndoe committed:
Update Polish translation.

+ Thank you rotrot!

johnjohndoe committed:
Use implicit Kotlin import.

johnjohndoe committed:
Merge pull request #713 from EventFahrplan/translations

Update Polish and Lithuanian translations.

johnjohndoe committed:
Apply automatic formatting.

johnjohndoe committed:
Improve color contrast of new session items in schedule changes screen.

johnjohndoe committed:
Merge pull request #715 from EventFahrplan/random-housekeeping

Random housekeeping (crash report, headline flickering, identical texts, schedule changes color contrast).

johnjohndoe committed:
Show room state in details screen (FOSDEM only).

+ The state of a room indicates how packed it is and whether it is worth
  going there or even evacuating it.
+ This feature relies on the room states API provided by FOSDEM and is
  therefore disabled by default for all events.
+ Room states API: https://api.fosdem.org
+ Room states library: https://github.com/EventFahrplan/roomstates

johnjohndoe committed:
Merge pull request #716 from EventFahrplan/fosdem-room-states

Show room state in details screen (FOSDEM only).

johnjohndoe committed:
Bump application version to v.1.69.0.

johnjohndoe committed:
Extract design system composables from "alarms" screen.

johnjohndoe committed:
Extract design system composables from "about" screen.

johnjohndoe committed:
Extract design system composables from "schedule changes" screen.

johnjohndoe committed:
Extract design system composables from "schedule statistics" screen.

johnjohndoe committed:
Extract design system composables from "search" screens.

johnjohndoe committed:
Extract design system composables from common composables.

johnjohndoe committed:
Restructure theme anatomy (EventFahrplanTheme, ColorSchemes).

+ Add Theme object to allow nested composables to access color values.
+ Docs: https://developer.android.com/develop/ui/compose/designsystems/anatomy

johnjohndoe committed:
Move IconVideoRecording composable into "designsystem" package.

johnjohndoe committed:
Move TopBar composable into "designsystem" package.

johnjohndoe committed:
Move HeaderSessionList composable into "designsystem" package.

johnjohndoe committed:
Move HeaderDayDate composable into "designsystem" package.

johnjohndoe committed:
Move NoData composable into "designsystem" package.

johnjohndoe committed:
Move Loading composable into "designsystem" package.

johnjohndoe committed:
Move TextClickable composable into "designsystem" package.

Commit:
Merge branch 'master' of https://github.com/EventFahrplan/EventFahrplan into glt-2025

# Conflicts:
#	.github/workflows/build.yaml
#	.travis.yml
#	README.md
#	app/build.gradle
#	database/src/main/java/info/metadude/android/eventfahrplan/database/sqliteopenhelper/SessionsDBOpenHelper.java

johnjohndoe committed:
Reduce double empty lines to single empty line.

+ Windows line endings are rendered as two separate empty lines when the
  text is rendered as Markdown. Hence, Windows line endings are converted
  to Unix line endings.
+ This does not fix line endings of sessions which have already been
  parsed and stored in the local database.
+ Affects "details" screen.

johnjohndoe committed:
Add default color for "multi_choice_background".

+ A product flavor can overwrite this color to customize its value.
+ "ccc38c3" and "cccamp2023" currently overwrite the color value.

johnjohndoe committed:
Make alarm minutes text on bell icon more readable.

+ This affects the alarms list screen.

johnjohndoe committed:
Add missing library name.

+ The "Room states" library was added here: 8a154435b5991da83f06c05830bfa5382c347f2d

johnjohndoe committed:
Use room-states-repositories v.1.0.1.

+ Fixes obfuscation issue caused by missing ProGuard rule.
+ Changelog: https://github.com/EventFahrplan/roomstates/releases/tag/v.1.0.1

johnjohndoe committed:
Use Gradle wrapper v.8.12.1.

+ Finally define distributionSha256Sum.
+ Changelog: https://docs.gradle.org/8.12.1/release-notes.html

johnjohndoe committed:
Document compileSdk requirement for activity-compose.

+ Changelog: https://developer.android.com/jetpack/androidx/releases/activity#1.10.0

johnjohndoe committed:
Use gradle-versions-plugin v.0.52.0.

+ Changelog: https://github.com/ben-manes/gradle-versions-plugin/compare/v0.51.0...v0.52.0.

johnjohndoe committed:
Use kotlin v.2.1.10 & ksp v.2.1.10-1.0.29.

+ Changelogs:
  https://github.com/JetBrains/kotlin/releases/tag/v2.1.10
  https://github.com/google/ksp/releases/tag/2.1.10-1.0.29

johnjohndoe committed:
Use Jetpack Compose UI 2025.01.01.

+ Changelogs:
  https://developer.android.com/develop/ui/compose/bom/bom-mapping
  https://developer.android.com/jetpack/androidx/releases/compose-material3#1.3.1

johnjohndoe committed:
Merge pull request #719 from EventFahrplan/project-updates

Update project setup (room-states-repositories 1.0.1, kotlin 2.1.10, ksp 2.1.10-1.0.29, Jetpack Compose UI 2025.01.01).

johnjohndoe committed:
Prefer CoroutineContext over CoroutineDispatcher in the public API.

+ Ref: https://code.cash.app/dispatchers-unconfined

johnjohndoe committed:
Add default color for "colorActionBar".

+ A product flavor can overwrite this color to customize its value.

johnjohndoe committed:
Ease browsing Material3 color scheme properties.

+ Apply the same sorting order to internal color scheme as
  being used by the respective Material3 properties.

johnjohndoe committed:
Wire more theme properties used by Material3 components within the app.

+ Remove used "about_horizontal_line" - rewire via "outlineVariant".
  Unused since: 34b1f3681463d0ba5845aec2cd3ffa2f33dcac8a
+ Related: 1e12b12425f620ade78ea0e36742fa26c375445b

johnjohndoe committed:
Remove unused import.

johnjohndoe committed:
Ease maintaining SessionChangeParametersFactoryTest.

johnjohndoe committed:
Encapsulate room name customization details within SessionPropertiesFormatter.

johnjohndoe committed:
Merge pull request #720 from EventFahrplan/random-housekeeping

Random housekeeping (ease maintenance, default colors, extend Compose theme, handle Windows line endings).

johnjohndoe committed:
Update Dutch translation.

+ Thank you @robinjanssens!

johnjohndoe committed:
Merge pull request #721 from EventFahrplan/translation-nl

Update Dutch translation.

johnjohndoe committed:
Facilitate simpler testing by introducing ContentDescriptionFormatter interface.

johnjohndoe committed:
Facilitate simpler testing by introducing FeedbackUrlComposition interface.

johnjohndoe committed:
Facilitate simpler testing by introducing SessionPropertiesFormatting interface.

johnjohndoe committed:
Use ContentDescriptionFormatting interface.

johnjohndoe committed:
Use FeedbackUrlComposition interface.

johnjohndoe committed:
Use SessionPropertiesFormatting interface.

johnjohndoe committed:
Clean up code.

+ Improve property and function names.
+ Remove unused import.

johnjohndoe committed:
Encode which markup language is used by a particular server backend.

+ Not used yet. This information is a preparation for the migration of
  the details screen from XML based layout to Compose UI.

johnjohndoe committed:
Migrate "session details" screen to Compose UI.

+ Replace XML-based screen with Compose UI setup.
+ The abstract and description texts are formatted as either HTML or
  Markdown depending on the markup language associated with the
  server backend of the event.
+ Introduce SessionDetailsParameterFactory and SelectedSessionParameterFactory
  to encapsulate screen logic. Add unit tests.
+ Text is now selectable and can be copied.
+ Use htmlconverter v.1.0.3. https://github.com/cbeyls/HtmlConverterCompose
+ Use multiplatform-markdown-renderer-m3 v.0.31.0. https://github.com/mikepenz/multiplatform-markdown-renderer/
+ Custom fonts are dropped for this screen.
+ Vertical scroll bar is missing. See https://developer.android.com/jetpack/androidx/compose-roadmap

Resolves #709

johnjohndoe committed:
Customize name of "Session Details" screen.

johnjohndoe committed:
Use Int#toDrawable convenience function.

johnjohndoe committed:
Add missing padding for favorites screen when shown on tablet in portrait mode.

johnjohndoe committed:
Cover SessionDetailsViewModel#roomStateMessage with unit tests.

+ Related: 8a154435b5991da83f06c05830bfa5382c347f2d

johnjohndoe committed:
Merge pull request #723 from EventFahrplan/random-housekeeping

Random housekeeping (Favorites padding, session details name, add interfaces, room state unit tests).

Commit:
Merge branch 'master' of https://github.com/EventFahrplan/EventFahrplan into glt-2025

johnjohndoe committed:
Merge pull request #724 from EventFahrplan/designsystem

Introduce design system.

johnjohndoe committed:
Resolve Lint warning.

johnjohndoe committed:
Merge pull request #725 from EventFahrplan/composify-session-details

Migrate "session details" screen to Compose UI.

johnjohndoe committed:
Update Polish translation.

+ Thank you rotrot!

johnjohndoe committed:
Merge pull request #726 from EventFahrplan/translations

Update Polish translation.

johnjohndoe committed:
Consolidate multiple XML layout files for the schedule time column.

+ Restructure to reduce complexity and nesting.

johnjohndoe committed:
Reduce layout files for schedule time column time text from 2 to 1.

johnjohndoe committed:
Drop unneeded dimension resources for schedule time column text.

+ values-xlarge will fallback to values-sw600dp or values-sw720dp.
+ values-port can be represented by values.

johnjohndoe committed:
Rename drawable to match name of related resources.

johnjohndoe committed:
Replace layer-list with single item with inset.

johnjohndoe committed:
Replace deprecated layout attribute "fill_parent" with "match_parent".

johnjohndoe committed:
Use same "android:fadingEdge" value for all schedule layout files.

+ Was never there, see main.xml at e4104abd9f232a28307a167ed8e39bd51ca0b93b.

johnjohndoe committed:
Use same "android:layout_width" value for all schedule layout files.

johnjohndoe committed:
Remove "android:layout_weight" attributes which have no effect from schedule layout files.

johnjohndoe committed:
Reduce complexity and duplication in "schedule" layout files.

+ Identified issues:
  - Extensive duplication: schedule.xml, schedule_land.xml, and
    schedule_land_large.xml are virtually identical.
  - Remove unnecessary indirection. The resource pointer XMLs add
    complexity without benefits.
  - Redundant resource qualifiers: Android's resource selection system
    can handle this more elegantly.

+ Former layout structure:

  schedule-related layout files:
  1. app/src/main/res/layout-port/schedule.xml (portrait layout)
  2. app/src/main/res/layout/schedule_land.xml (landscape layout)
  3. app/src/main/res/layout/schedule_land_large.xml (large landscape layout)

  and resource files that reference them:
  4. app/src/main/res/values-land/schedule.xml (points to schedule_land.xml)
  5. app/src/main/res/values-xlarge/schedule.xml (points to schedule_land_large.xml)
  6. app/src/main/res/values-sw600dp/schedule.xml (points to schedule_land_large.xml)

+ Layout files (1, 2, 3) are identical.

johnjohndoe committed:
Improve "time column time text" layout file name.

johnjohndoe committed:
Improve "time column time text" identifier.

johnjohndoe committed:
Merge pull request #727 from EventFahrplan/housekeeping-schedule-time-column

Consolidate multiple XML layout files for the schedule time column.

johnjohndoe committed:
Merge pull request #728 from EventFahrplan/housekeeping-schedule-layout-files

Reduce complexity and duplication in "schedule" layout files.

Commit:
Merge branch 'master' of https://github.com/EventFahrplan/EventFahrplan into glt-2025

Commit:
Bumps the Java Version because of Gradle complaining.
Bumps the DB version to clean out the GLT24 schedule
Bumps the version number of the app

Commit:
Bumps the Java Version in GH WF files.

Commit:
Revert the Java Version in GH WF files so that hopefully the unit-test pass.

johnjohndoe commented:
What is the error you noticed when using Java 17?

Commit:
Revert the Java Version in GH WF files so that hopefully the unit-test pass.

Rufus125 committed:
Merge branch 'glt-2022' into glt-2025

Rufus125 commented:
> What is the error you noticed when using Java 17?

Android Studio complained during compile that the Java Version (17) was not the same as Kotlin (21), but I guess it works again after an update of Android Studio.

johnjohndoe commented:
It was likely the JDK setting inside Android Studio.

mgorfer committed:
Update Dates for Info Page

Rufus125 committed:
Merge pull request #1 from mgorfer/glt-2025

Review the PR for GLT 2025
